src/ast/codegen_llvm.cpp

```diff
@@ -903,7 +904,14 @@ void CodegenLLVM::visit(FieldAccess &acc)
   assert(type.type == Type::cast);
   acc.expr->accept(*this);
 
-  auto &field = bpftrace_.structs_[type.cast_type].fields[acc.field];
+  std::string cast_type = type.is_tparg ? tracepoint_struct_ : type.cast_type;
+  Struct &cstruct = bpftrace_.structs_[cast_type];
+
+  type = SizedType(type);
```

Comment:
What's this here for? Is it a mistake?

Reply:
Might be a mistake, but I'm not sure (it's been some time since I've worked on this).